sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierSerializerTest.cs

```diff
@@ -17,8 +17,11 @@ public void MissingProperty_DeserializerThrows()
                 new CommunicationIdentifierModel(CommunicationIdentifierKind.CommunicationUser), // Missing Id
                 new CommunicationIdentifierModel(CommunicationIdentifierKind.CallingApplication), // Missing Id
                 new CommunicationIdentifierModel(CommunicationIdentifierKind.PhoneNumber) { Id = "some id" }, // Missing PhoneNumber
-                new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser) { Id = "some id", MicrosoftTeamsUserId = "some id" }, // Missing IsAnonymous
-                new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser) { Id = "some id", IsAnonymous = true }, // Missing MicrosoftTeamsUserId
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.PhoneNumber) { PhoneNumber = "+12223334444" }, // Missing Id
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser) { Id = "some id", MicrosoftTeamsUserId = "some id", Cloud = CommunicationCloudEnvironmentModel.Public }, // Missing IsAnonymous
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser) { Id = "some id", IsAnonymous = true, Cloud = CommunicationCloudEnvironmentModel.Public }, // Missing MicrosoftTeamsUserId
+                new CommunicationIdentifierModel(CommunicationIdentifierKind.MicrosoftTeamsUser) { MicrosoftTeamsUserId = "some id", IsAnonymous = true, Cloud = CommunicationCloudEnvironmentModel.Public }, // Missing Id
```

Comment:
But in the constructor of MicrosoftTeamsUserIdentifier, id is optional. Should we throw in this case?

Reply:
Even though the id for MicrosoftTeamsUserIdenfier is optional to be set by the developers, it is mandatory to be sent by the services. Hence deserialization assumes that it is always present